### ISSUE TITLE
nightly cargo needed

### ISSUE THREAD

**Issue by holtzermann17** [lapce/lapce#572]
```
joe@Proteus:~/lapce$ cargo build --release
error: failed to parse manifest at `/home/joe/lapce/Cargo.toml`

Caused by:
  feature `named-profiles` is required

  this Cargo does not support nightly features, but if you
  switch to nightly channel you can add
  `cargo-features = ["named-profiles"]` to enable this feature
```

Fixed by `rustup toolchain install nightly && rustup default nightly` and rerun above.

**Comment by informaticalex** [comment]
Custom `named-profiles` are a feature of [Cargo version 1.57](https://doc.rust-lang.org/cargo/reference/unstable.html?highlight=named-profile#custom-named-profiles). Regardless of whether a custom profile is specified, building with Cargo < 1.57 will show the above error.

As mentioned, upgrading Cargo to nightly, or 1.57+, fixes this.

Tested on [Fedora Workstation 36](https://getfedora.org/) using [this method](https://users.rust-lang.org/t/how-can-i-revert-to-older-version-of-rust/20497) to downgrade Cargo/Rust.

**Comment by informaticalex** [comment]
Related to the above, Cargo version 1.57-1.59 will have a different error:

```
error: failed to load manifest for workspace member `C~\lapce\lapce-ui`

Caused by:
  failed to load manifest for dependency `lapce-core`

Caused by:
  failed to parse manifest at `C:\Users\mieza\lapce\lapce-core\Cargo.toml`

Caused by:
  namespaced features with the `dep:` prefix are only allowed on the nightly channel and requires the `-Z namespaced-features` flag on the command-line
```

Lapce is using _optional_ dependencies, a feature implemented in [Cargo 1.60](https://github.com/rust-lang/cargo/blob/master/CHANGELOG.md#cargo-160-2022-04-07).

Cargo must be version 1.60+ for lapce to build.

**Comment by panekj** [comment]
Fixed in https://github.com/lapce/lapce/pull/573
